server/kv.go

```diff
@@ -474,6 +482,47 @@ func (store store) DeleteUserInfo(ji Instance, mattermostUserId string) (returnE
 	return nil
 }
 
+var reHexKeyFormat = regexp.MustCompile("^[[:xdigit:]]{32}$")
+
+func (store store) CountUsers(ji Instance) (int, error) {
+	count := 0
+	for i := 0; ; i++ {
+		keys, appErr := store.plugin.API.KVList(i, listPerPage)
+		if appErr != nil {
+			return 0, appErr
+		}
+
+		for _, key := range keys {
+			// User records are not currently prefixed. Consider any 32-hex key.
+			if !reHexKeyFormat.MatchString(key) {
+				continue
+			}
+
+			var data []byte
+			data, appErr = store.plugin.API.KVGet(key)
+			if appErr != nil {
+				return 0, appErr
+			}
+			v := map[string]interface{}{}
+			err := json.Unmarshal(data, &v)
+			if err != nil {
+				// Skip non-JSON values.
+				continue
+			}
+
+			// A valid user record?
+			if v["Settings"] != nil && (v["accountId"] != nil || v["name"] != nil && v["key"] != nil) {
```

Comment:
Is it possible that these values are empty string instead of `nil`?

Reply:
Testing for "nil" distinguishes Users from other structs where the fields are never present.